src/server/routes/actions.spec.ts

```diff
@@ -0,0 +1,279 @@
+      expect(mockHandleRequest).toHaveBeenCalledWith(
+        expect.objectContaining({ external: true, address: "example2", recaptcha: "captcha2" }),
+      );
+      expect(res.status).toBe(500);
+      expect(res.body.error).toBe(error);
+    });
+
+    test("should report error on internal error", async () => {
+      mockHandleRequest.mockRejectedValueOnce("random error in /web");
+      const res = await request(app).post("/drip/web").send({ address: "example3", recaptcha: "captcha3" });
+      expect(res.status).toBe(500);
+      expect(res.body.error).toBe("Operation failed.");
+      expect(mockLoggerError).toHaveBeenCalledWith("random error in /web");
+    });
+  });
+
+  describe("/drip/bot", () => {
+    beforeEach(() => {
+      mockConfig.mockImplementation((type: string) => {
+        if (type === "EXTERNAL_ACCESS") {
+          return false;
+        }
+        return "generic";
+      });
+    });
+
+    test("should fail with no address", async () => {
+      const res = await request(app).post("/drip/bot");
+      expect(res.body.error).toBe(parameterError("address"));
+      expect(res.status).toBe(400);
+    });
+
+    test("should fail if external access is enabled", async () => {
+      mockConfig.mockImplementation((type: string) => {
+        if (type === "EXTERNAL_ACCESS") {
+          return true;
+        }
+        return "generic";
+      });
+
+      const res = await request(app).post("/drip/bot").send({ address: "example" });
+      expect(res.body.error).toBe("Endpoint unavailable");
+      expect(res.status).toBe(503);
+    });
+
+    test("should fail with no amount", async () => {
+      const res = await request(app).post("/drip/bot").send({ address: "example" });
+      expect(res.body.error).toBe(parameterError("amount"));
+      expect(res.status).toBe(400);
+    });
+
+    test("should fail with no sender", async () => {
+      const res = await request(app).post("/drip/bot").send({ address: "example", amount: "100" });
+      expect(res.body.error).toBe(parameterError("sender"));
+      expect(res.status).toBe(400);
+    });
+
+    test("should fail with no sender", async () => {
+      const res = await request(app).post("/drip/bot").send({ address: "example1", amount: "100", sender: "sender1" });
+      expect(res.status).toBe(200);
+      expect(mockHandleRequest).toHaveBeenCalledWith(
+        expect.objectContaining({ external: false, address: "example1", amount: "100", sender: "sender1" }),
+      );
+    });
+
+    test("should report error on internal error", async () => {
+      mockHandleRequest.mockRejectedValueOnce("random error in /bot");
+      const res = await request(app).post("/drip/bot").send({ address: "example", amount: "0", sender: "sender" });
+      expect(res.status).toBe(500);
+      expect(res.body.error).toBe("Operation failed.");
+      expect(mockLoggerError).toHaveBeenCalledWith("random error in /bot");
+    });
+  });
+});
+
+describe("/drip tests", () => {
```

Comment:
Since https://github.com/paritytech/substrate-matrix-faucet/issues/215 is done, we can remove those tests and this endpoint.

Reply:
Let's merge it for now and remove it in the next PR: #217

This way, if, for some reason, we need to revert, we can use the tests.

What do you think?

Reply:
Yes, agree 👍